src/main/java/io/phdata/streamliner/schemadefiner/model/TableDiff.java

```diff
@@ -38,8 +39,61 @@ public TableDiff(
     this.existsInSource = existsInSource;
   }
 
-  public boolean allChangesAreCompatible(){
-    return columnDiffs.stream().allMatch(c -> c.getIsAdd());
+  public boolean allChangesAreCompatible(scala.collection.immutable.Map<String, scala.collection.immutable.Map<String, String>>
+                                                 typeMapping) {
+    Map<String, Map<String, String>> javaTypeMap = JavaHelper.convertScalaMapToJavaMap(typeMapping);
+    boolean flag = true;
+    for (ColumnDiff colDiff : columnDiffs) {
+      if (colDiff.getIsDeleted()) {
+        flag = false;
+        break;
+      } else if (colDiff.getIsUpdate()) {
+        ColumnDefinition currDef = colDiff.getCurrentColumnDef();
+        ColumnDefinition prevDef = colDiff.getPreviousColumnDef();
+
+        // currently datatype increase is implemented only for snowflake string data type
+        if (!isSnowflakeStringDataType(javaTypeMap, currDef)
+            || currDef.getPrecision() == prevDef.getPrecision()) {
```

Comment:
I don't get this part:
currDef.getPrecision() == prevDef.getPrecision()

Reply:
ColumnDiff isUpdate becomes true if any variable of the ColumnDefinition changes.
Since we are checking increment of column length only, precision is being checked here. 
Hence if isUpdate is true but precision is not changed, we will be returning false. This makes change not compatible and it will be added in templateContext for investigation.